lib/blockscore/actions/retrieve.rb

```diff
@@ -12,7 +12,7 @@ module Actions
     module Retrieve
       module ClassMethods
         def retrieve(id, options = {})
-          fail ArgumentError if id.empty?
+          fail ArgumentError,'ID must be supplied' if id.nil? || id.empty?
```

Comment:
Instead of checking for what the value isn't (blacklisting), what about checking for what the value should be? (whitelisting)

You could do something like:

```ruby
fail ArgumentError, 'ID must be be a natural number' unless id.to_s =~ /\A\d+\z/
```

I would tend to factor the regexp out into a constant though. I might also look at the number specifically. Would `0000` be a valid id? Is `01`? I might start off with something strict and then loosen it over time rather than start off weaker and tighten it -- the former is much easier than the latter.

In this specific case I might even recommend something like: `/\A[1-9]\d*\z/`

This prevents numbers with leading zeros. I don't know if there's a maximum length or if it's unbounded, but that could be factored in, eg: `/\A[1-9]\d{,5}\z/`

Also note the usage of `\A` and `\z`. Lots of people in ruby-land will reach for `^` and `$` in their regexps. 99.9% of the time __this is a bug__, consider:

```
"a\n1\nb" =~ /^\d+$/
=> 2

"a\n1\nb" =~ /\A\d+\z/
=> nil
```

If you have validation code using `^` and `$` in the regexp, there's a good chance invalid data can sneak by. I have found this to be extremely common in ruby apps.